src/python/pants/backend/python/goals/repl.py

```diff
@@ -155,7 +155,7 @@ async def create_ipython_repl_request(
         "PEX_EXTRA_SYS_PATH": os.pathsep.join(chrooted_source_roots),
     }
 
-    return ReplRequest(digest=merged_digest, args=args, extra_env=extra_env)
+    return ReplRequest(digest=merged_digest, args=args, extra_env=extra_env, run_in_workspace=False)
```

Comment:
@stuhood  / @Eric-Arellano: I had to add `run_in_workspace=False` here to actually get tests to pass. Despite constructing a `Digest`, the core repl goal was not passing that `Digest` into `InteractiveProcess` until this PR.

Reply:
Hm, we had them running in the workspace so that you can do things like `with open()` even if the file is not declared with Pants. This is probably not a good change.

I'm confused why it would only start failing now? Is it the append-only-caches codepath is now being triggered maybe? That wasn't wired up until now due to the changes in `core/goals/repl.py`.

Reply:
I've broken out the changes to the core `repl` rules to make the issue clearer. Let's discuss on https://github.com/pantsbuild/pants/pull/13599.

Reply:
This change to the Python repl rules has been reverted since #13599 landed.